crypto-core/src/main/java/com/palantir/crypto2/io/CryptoStreamFactory.java

```diff
@@ -91,11 +94,26 @@ static OutputStream encrypt(OutputStream output, KeyMaterial keyMaterial, String
         try {
             return createApacheEncryptedStream(output, keyMaterial);
         } catch (IOException e) {
-            log.warn("Unable to initialize cipher with OpenSSL, falling back to JCE implementation");
+            warningLog(e);
             return createDefaultEncryptedStream(output, keyMaterial, algorithm);
         }
     }
 
+    /** To avoid spamming logs with exceptions, we only log the exception at most once per minute. */
+    private static void warningLog(IOException exception) {
+        String message = "Unable to initialize cipher with OpenSSL, falling back to JCE implementation "
+                + "- see github.com/palantir/hadoop-crypto";
```

Comment:
Can we just log `OpenSSL not found or incompatible version - to fix see github.com/palantir/hadoop-crypto` and not even worry about logging the exception every 60 seconds?

Reply:
I think we'll want to be able to see the exception on the day when it doesn't work for some new reason, e.g., some new incompatibility with commons-crypto when we upgrade OpenSSL past 1.1.

(I was a fan of the "log the stacktrace at debug" idea, btw -- the once a minute thing is still gonna be quite noisy.)

Reply:
> not even worry about logging the exception every 60 seconds

I really want a way to be able to see this exception, even just to validate that they're all the same across the fleet. We currently _think_ they're all the same, but can't easily validate this.

> the "log the stacktrace at debug" idea

The reason I don't like debug is that when I opened up Aries, there were >20 services hitting this problem, and I'm not going to spend time to go through and open a bunch of cdconfig PRs to dial them up to debug. I want to be able to quickly assess whether we're hitting something new, or this is the same old problem we already know how to solve.

> once a minute thing is still gonna be quite noisy

Is once a minute really going to be that noisy? Looking at my sls-logging dashboard, I can see a lot of services log >200lines per second... going from 12,000 -> 12,001 per minute seems like not that bad? Alternatively I can happily change this to log the exception just the first time it hits, assuming it'll be the same stacktrace for the lifetime of the service?

Reply:
Re: the once a minute idea being noisy, it's just that stacktraces are big, so the log bloat is exaggerated by that.

Logging the exception only the first time it hits seems legit to me.

Reply:
I'm also cool with logging it the first time.

Reply:
Ok have updated